### PR TITLE
Bitsight collector and parser

### PR DESCRIPTION
**Pull request by pedromreis**

### CODE REVIEW

**Comment by pedromreis** [comment]
Humm, sorry for this, I was trting to update my git...
:)

File: intelmq/bots/parsers/bitsight/parser.py [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+               event.add('source.geolocation.country', value)
+            event.add("raw", json.dumps(raw_report, sort_keys=True))
+            self.send_message(event)
+        self.acknowledge_message()
+
+if __name__ == "__main__":
+    bot = BitsightParserBot(sys.argv[1])
+    bot.start()
+
+
```
Comment (sebix) [inline intelmq/bots/parsers/bitsight/parser.py]: Please just one newline at the end.

File: intelmq/bots/parsers/bitsight/parser.py [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+            self.acknowledge_message()
+            return
+        raw_report = {}
+        raw_report = json.loads(utils.base64_decode(report.get('raw')))
+        for key in raw_report:
+            extra = {}
+            event = Event(report)
+            value = raw_report[key]
+            if key == "_ts":
+               print('ts = ' + str(value))
```
Comment (sebix) [inline intelmq/bots/parsers/bitsight/parser.py]: Please remove debugging prints

File: intelmq/bots/parsers/bitsight/parser.py [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+               #event.add('time.source', value)
```
Comment (sebix) [inline intelmq/bots/parsers/bitsight/parser.py]: Events without `time.source` are useless.

File: intelmq/bots/parsers/bitsight/parser.py [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+                   event.add('source.port', value["remote_port"])
+               if "server_addr" in value:
+                   event.add('destination.ip', value["server_addr"])
+               if "server_port" in value:
+                   event.add('destination.port', value["server_port"])
+               if "server_name" in value:
+                   try:
+                       event.add('destination.fqdn', value["server_name"])
+                   except:
+                       event.add('destination.fqdn', "")
```
Comment (sebix) [inline intelmq/bots/parsers/bitsight/parser.py]: An empty string is not valid.

File: intelmq/bots/parsers/bitsight/parser.py [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+               if "remote_port" in value:
+                   event.add('source.port', value["remote_port"])
+               if "server_addr" in value:
+                   event.add('destination.ip', value["server_addr"])
+               if "server_port" in value:
+                   event.add('destination.port', value["server_port"])
+               if "server_name" in value:
+                   try:
+                       event.add('destination.fqdn', value["server_name"])
+                   except:
```
Comment (sebix) [inline intelmq/bots/parsers/bitsight/parser.py]: You only want to catch `intelmq.lib.exceptions.InvalidValue`

File: intelmq/bots/parsers/bitsight/parser.py [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+class BitsightParserBot(Bot):
+
+    def process(self):
+        report = self.receive_message()
+        if report is None or not report.contains('raw'):
+            self.acknowledge_message()
+            return
+        raw_report = {}
+        raw_report = json.loads(utils.base64_decode(report.get('raw')))
+        for key in raw_report:
```
Comment (sebix) [inline intelmq/bots/parsers/bitsight/parser.py]: Better do `for key, value in raw_report.items()` and drop the `value=` line.

File: ok [hunk 7]
```diff
@@ -0,0 +1,8 @@
+kjhjk
```
Comment (sebix) [inline ok]: What is this file for?

File: intelmq/bots/parsers/bitsight/parser.py [hunk 8]
```diff
@@ -0,0 +1,74 @@
...
+                       event.add('destination.fqdn', value["server_name"])
+                   except:
+                       event.add('destination.fqdn', "")
+               if "request_method" in value:
+                  extra['request_method'] = value["request_method"]
+               if extra:
+                  event.add('extra', extra)
+            if key == "_geo_env_remote_addr.country_name":
+               event.add('source.geolocation.country', value)
+            event.add("raw", json.dumps(raw_report, sort_keys=True))
```
Comment (sebix) [inline intelmq/bots/parsers/bitsight/parser.py]: You can use `report.get('raw')` here.

File: intelmq/bots/BOTS [hunk 9]
```diff
@@ -1,5 +1,15 @@
 {
     "Collector": {
+
+        "bitsight":{
+        "module.intelmq.bots.collectors.bitsight.collector",
+        "parameters":{
+             "feed":"Bitsight",
+             "http_url": "http://alerts.bitsighttech.com:8080/stream?key=",
+             "api_key": "<insert your api key>",
```
Comment (sebix) [inline intelmq/bots/BOTS]: To be consistent with other bots, remove this option and add the `<api>` string to the URL

File: intelmq/bots/BOTS [hunk 10]
```diff
@@ -1,5 +1,15 @@
 {
     "Collector": {
+
+        "bitsight":{
```
Comment (sebix) [inline intelmq/bots/BOTS]: This file is sorted, and no empty lines are allowed. You can use the `test_conf.py` script to validate the file.

**Comment by sebix** [comment]
* Move your commits to a separate branch, working on master is highly discouraged
* Rebase on upstream master
* Write tests for the parser, see for examples: https://github.com/certtools/intelmq/blob/master/intelmq/tests/bots/parsers/malwaredomainlist/test_parser.py
* Write a dummy test for the collector, see e.g. https://github.com/certtools/intelmq/blob/master/intelmq/tests/bots/collectors/http/test_collector.py

File: intelmq/bots/BOTS (hunk 10 above)
Comment (sebix) [inline intelmq/bots/BOTS]: Also, please adapt the bot name (e.g. `Bitsight Alert Stream`), collector and parser name should be same/similar.

**Comment by pedromreis** [comment]
Sorry, this was not supposed to get here....  not now at least, the code is not working yet.
But I thank you for the comments. I'll implement them :)

**Comment by sebix** [comment]
Obsoleted by certtools/intelmq#484

**Pull request closed without merging**